### DISCUSSION TITLE
HIP Windows vs VStudio 2019 Intellisense

### DISCUSSION THREAD

**Discussion by tomashomecat** [ROCm/ROCm#3521] [Q&A]
Hi All
I'm a new HIP user so ask don't blame me for wasted time. The system - Win10, VS2019. AMD 7840hs (gfx1103). The problem - HIP samples shows THOUSANDS Intellisense errors (on screenshot). so - What i did wrong? and how to resolve? I can builds and run samples, i know i can "disable Intellisense errors" but - i want to use this feature.
Thanks for advance for reading and understanding.
![hipIntelErr](https://github.com/user-attachments/assets/809fccfe-8e91-49fb-8dc0-5c6a92a979da)

**Comment by schung-amd** [comment]
Hi @tomashomecat, are you still experiencing this?

**Reply by tomashomecat** [reply]
yes of course. the problem persists and no solution found. I disabled intellisense errors, because it able to builds and execute.

**Reply by schung-amd** [reply]
What build of Visual Studio do you have? Have you followed the steps at https://rocm.docs.amd.com/projects/hip-vs/en/latest/index.html?

**Reply by tomashomecat** [reply]
Hi schung,
Microsoft Visual Studio Community 2019 Version 16.11.42
.Net desktop Development
.Net Framework 4.8.04084
i use LAST official HIP 6.1.2 , i dont understand why at Your link mentioned v6.2  , i cannot find HIP with this version.
and at last main issue regarding Your link - there is NO files to download, as i understand this project is deprecated, and i see only "http error 404" page,Ii installed project templates which are coming with HIP 6.1 distributive
as i said the problem with Intellisense persists and looks like it is forever.

Best regards!

**Reply by schung-amd** [reply]
The docs mention 6.2 as that is the latest mainline version; on Windows the latest HIP SDK version is currently 6.1.2 as you note, but the docs should still be relevant. That being said, I was unaware that repo is private now, and the docs shouldn't be pointing there anymore; will raise this to the internal team, sorry about that.

You might already have the extension installed from HIP SDK; can you check to see if you have a HIP extension installed/enabled in the VS extension manager?

**Reply by tomashomecat** [reply]
of course, it was installed from 6.1.2 distib. extension manager shows - AMD HIP Toolchain 6.1 version 6.1.0.4
just 1 question - is it only my problem and all other known AMD Windows HIP SDK installations havent this issue? i am trying to understand what went wrong during installations?

**Reply by schung-amd** [reply]
I haven't seen other reports of this and personally didn't run into this when I tried VS2019 + HIP SDK earlier this year, so I do suspect a configuration or caching issue somewhere and not necessarily something broken on our end; I'll try to reproduce your issue to confirm. Thanks for reporting this!

**Reply by tomashomecat** [reply]
Hi schung
1. some links to problems looking like similar to mine. i hope it will help You to understand whats going on
https://stackoverflow.com/questions/78953829/hip-gpu-programming-in-visual-studio-2022-amd-unresolved-externals-blockidx
https://github.com/ROCm/ROCm/discussions/2492
2. and again - at Your opinion and You are confident that this problem should NOT exists, and i am experiencing this issue only because i have some configuration-installation discrepancies on my PC?

**Reply by schung-amd** [reply]
Yeah, I think there's something wrong here and these errors should not be showing up, so I'm trying to find the root cause and a solution for you. Thanks for those links, the second seems relevant here. I'll let you know if I'm able to reproduce what you're seeing.

**Reply by schung-amd** [reply]
Do you see this many errors when starting a new project with the sample Matrix Transpose template? With HIP SDK 6.1 there are two templates called "AMD HIP 6.1 Runtime", one stating that it is a matrix transpose example in the description.

**Reply by tomashomecat** [reply]
Hi schung
yes of course, i see Intellisense errors in project based on this template. I am not too much familiar with C++ so when possible i always use template for such projects.
![image](https://github.com/user-attachments/assets/4934387e-36bd-4eb6-8c38-ab612abffdb8)

**Reply by schung-amd** [reply]
Thanks, I was able to repro this on a 7840u in VS2022, so it's not just your configuration. I'll take a look to see why this is happening.

My initial suspicion is that this is at least in part because we don't support gfx1103; see https://rocm.docs.amd.com/projects/install-on-windows/en/latest/reference/system-requirements.html. While I can build and run the matrix transpose template despite the Intellisense errors, it also does not give correct results. As an additional data point, I do not see the Intellisense warnings or wrong results with a W7900, which is supported.

**Comment by tomashomecat** [comment]
Hi schung
Thank You for spending Your time to investigate this case. My opinion - it is not looking as reaction on incompatible GPU. because i dont see any caution about incompatible GPU from HIP system, i am able successfully build and execute code and only at this stage system is reporting that it cannot find compatible GPU. At this stage of my test-work i disabled Intellisense warnings, and just studying the source code in examples. i intend to reinstall Windows system when i will have compatible GPU.
![image](https://github.com/user-attachments/assets/2480f5f7-9188-4604-a2cd-f6973af91240)

**Reply by schung-amd** [reply]
I'm able to get correct results for this example after manually adding "gfx1103" to the list of offload architectures in the project properties under `Configuration Properties > General [AMD HIP for clang]` even though there's no official support for gfx1103. This doesn't resolve the Intellisense issue, though, and I don't think this means everything in the HIP SDK will necessarily work on gfx1103.

**Comment by tomashomecat** [comment]
Hi schung
i am still searching for the root cause of issue - as i understand it is because Intellisense and builder somehow use different compilers. 
https://stackoverflow.com/questions/77790798/intellisense-is-showing-errors-from-standard-library-while-using-stdclatest
in another cases - some people have reverse situation - Intellisense shows no error but compiling fails.. i intend to find where in VStudio configs i can set these things.
